swap_meet/vendor.py

```diff
@@ -1,2 +1,60 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
+
+# Wave 1
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+        self.item = Item()
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        else: 
+            self.inventory.remove(item)
+            return item
+
+# Wave 2
+    def get_by_category(self, category):
+        inventory_items = []
+        for item in self.inventory:
+            if item.category == category:
+                inventory_items.append(item)
+        return inventory_items
+
+# Wave 3
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item in self.inventory and my_item not in vendor.inventory and their_item in vendor.inventory and their_item not in self.inventory:
+            self.inventory.remove(my_item)
+            vendor.inventory.append(my_item)
+            vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        else:
+            return False
+
+# Wave 4
+    def swap_first_item(self, vendor):
+        if len(self.inventory) == False or len(vendor.inventory) == False:
+            return False
+        else: 
+            self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
+            return True
+
+# Wave 6
+    def get_best_by_category(self, category = ""):
+        if len(self.get_by_category(category)) == False:
+            return None
+        else:
+            best_by_category = max(self.get_by_category(category), key = lambda item : item.condition)
+            return best_by_category
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        if len(self.inventory) == False or len(other.inventory) == False:
```

Comment:
Here again, be careful of how you're checking for empty lists. The more pythonic way would be
```python
        if not self.inventory or not other.inventory:
```

That being said, we don't really need to check whether the lists are empty at all! You made great use of some of your other methods, which already do the safety checks you need, and handle errors in the same way.

Consider what happens if one of the inventories doesn't have an item of the desired kind. Those calls would return `None` for the best item in that category. If we then try to swap none in `swap_items`, it wouldn't be found in the inventory, and the function would return `False` just as desired!